pkg/controller/core/workload_controller.go

```diff
@@ -525,9 +557,38 @@ func (h *resourceUpdatesHandler) queueReconcileForPending(ctx context.Context, _
 		wlCopy := w.DeepCopy()
 		log := log.WithValues("workload", klog.KObj(wlCopy))
 		log.V(5).Info("Queue reconcile for")
-		h.r.adjustResources(log, wlCopy)
+		h.r.adjustResources(ctx, wlCopy)
 		if !h.r.queues.AddOrUpdateWorkload(wlCopy) {
 			log.V(2).Info("Queue for workload didn't exist")
 		}
 	}
 }
+
+// podTemplateHandler signals the controller to reconcile the Workload associated
+// to the podtemplate in the event.
+// Since the events come from a channel Source, only the Generic handler will
+// receive events.
+type podTemplateHandler struct{}
+
+func (h *podTemplateHandler) Create(context.Context, event.CreateEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *podTemplateHandler) Update(_ context.Context, e event.UpdateEvent, q workqueue.RateLimitingInterface) {
```

Comment:
Uhm... I think we don't need both a podTemplateHandler and a PodTemplate Reconciler.

Just the PodTemplateHandler should be enough (should be similar to the existing resourceUpdatesHandler)